qemu/tests/scsi_mq_check.py

```diff
@@ -0,0 +1,71 @@
+import logging
+import re
+
+from virttest import error_context
+from virttest import utils_test
+from virttest import utils_misc
+
+
+def get_mq_from_guest(session, params, cmd, pattern):
+    """
+    Get multi queue number from guest.
+
+    :param session: VM session
+    :param params: Dictionary with the test parameters
+    :param cmd: command to get queue info
+    :param pattern: Get the queue number from queue info
+    """
+    status, output = session.cmd_status_output(cmd)
+    if status != 0:
+        return output
+    if params.get("os_type") == "windows":
+        try:
+            queues = int(re.search(pattern, output, re.I | re.M).group(1))
+        except IndexError:
+            return None
+    if params.get("os_type") == "linux":
+        queues = int(len(re.findall(pattern, output, re.I | re.M)))
+    if queues:
+        return queues
+    else:
+        return None
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Check device multi-queue function enabled.
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment
+    """
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    timeout = float(params.get("login_timeout", 240))
+    driver_name = params.get("driver_name")
+    session = vm.wait_for_login(timeout=timeout)
+    num_queues = params.get("num_queues")
+    pattern = params.get("pattern")
+    check_queue_cmd = params.get("check_queue_cmd")
```

Comment:
if `driver_name`, `num_queues` and `check_queue_cmd` are required params, better to use `pattern = params.["pattern"]` to indicate that.
Is `num_queues` defined in another cfg?

Reply:
ok, thanks, will update it, and for num_queues is defined by --customsparams="num_queues=4", it will be defined manually.